File: 258-add-digits/258-add-digits.cpp
class Solution {
public:
    int addDigits(int num) {
        int sum = 0;
        if(num<10) return num;
        while(num>0)
        {
            sum += num%10;
            num = num/10;
        }
        sum = sum%9==0?9:sum%9;
        return sum;
    }
};

File: 258-add-digits/258-add-digits.js
/**
 * @param {number} num
 * @return {number}
 */
var addDigits = function(num) {
    let sum = 0;
    if(num<10) return num;
    while(num > 0)
    {
        sum+=num%10;
        num=Math.floor(num/10);
    }
    // console.log(sum);
    return sum%9==0?9:sum%9;
};